validation/barotropic_gyre/barotropic_gyre.jl

```diff
@@ -0,0 +1,124 @@
+# # Barotropic gyre
+
+using Oceananigans
+using Oceananigans.Grids
+
+using Oceananigans.Coriolis:
+    HydrostaticSphericalCoriolis,
+    VectorInvariantEnergyConserving,
+    VectorInvariantEnstrophyConserving
+
+using Oceananigans.Models.HydrostaticFreeSurfaceModels:
+    HydrostaticFreeSurfaceModel,
+    VectorInvariant,
+    ExplicitFreeSurface
+
+using Oceananigans.TurbulenceClosures: HorizontallyCurvilinearAnisotropicDiffusivity
+using Oceananigans.Utils: prettytime, hours
+using Oceananigans.OutputWriters: JLD2OutputWriter, TimeInterval, IterationInterval
+
+using Statistics
+using JLD2
+using Printf
+using GLMakie
+
+Nx = 120
+Ny = 60
+
+# A spherical domain
+grid = RegularLatitudeLongitudeGrid(size = (Nx, Ny, 1),
+                                    longitude = (-30, 30),
+                                    latitude = (-15, 45),
+                                    z = (-4000, 0))
+
+free_surface = ExplicitFreeSurface(gravitational_acceleration=0.1)
+
+coriolis = HydrostaticSphericalCoriolis(scheme = VectorInvariantEnstrophyConserving())
+
+surface_wind_stress_parameters = (τ₀ = 5e-4,
+                                  Lϕ = grid.Ly,
+                                  ϕ₀ = 15)
+
+surface_wind_stress(λ, ϕ, t, p) = - p.τ₀ * cos(2π * (ϕ - p.ϕ₀) / p.Lϕ)
+
+surface_wind_stress_bc = BoundaryCondition(Flux,
+                                           surface_wind_stress,
+                                           parameters = surface_wind_stress_parameters)
+
+u_bcs = UVelocityBoundaryConditions(grid, top = surface_wind_stress_bc)
+                                        
+model = HydrostaticFreeSurfaceModel(grid = grid,
+                                    architecture = CPU(),
+                                    momentum_advection = VectorInvariant(),
+                                    free_surface = free_surface,
+                                    coriolis = coriolis,
+                                    boundary_conditions = (u=u_bcs,),
+                                    buoyancy = nothing,
+                                    closure = HorizontallyCurvilinearAnisotropicDiffusivity(νh=1e2, κh=1e2))
+
+g = model.free_surface.gravitational_acceleration
```

Comment:
Not suggesting anything should be changed here but now that I read this, it does seem odd to have `gravitational_acceleration` in `model.free-surface`.  Something to discuss elsewhere.

Reply:
We're trying to store all locally-used properties locally. But there's definitely also a need to specify "global" physical parameters somewhere. We're working on a solution for that...